src/main/java/io/seqera/tower/cli/commands/data/links/SearchOption.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021-2023, Seqera.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.seqera.tower.cli.commands.data.links;
+
+import io.seqera.tower.cli.utils.data.DataLinkProvider;
+import picocli.CommandLine;
+
+import java.util.List;
+
+public class SearchOption {
+
+    @CommandLine.Option(names = {"-n", "--name"}, description = "Show only data links with names that start with the given word.")
+    public String startsWith;
+
+    @CommandLine.Option(names = {"-r", "--region"}, description = "Show only data links belonging to given region")
+    public String region;
+
+    @CommandLine.Option(names = {"-p", "--providers"}, description = "Show only data links belonging to given providers. [aws,azure,google]")
```

Comment:
I think specifying the option like this expects multiple values to passed with multiple options, e.g. `-p aws -p azure` so providers doesn't really make sense as the option name. If you want to have a single option, you can add a split argument to the Option annotation, see https://picocli.info/#_split_regex

Reply:
Cool, I tried It and when invalid:
```bash
tw data-links list -p api,aws
```
then returned:
```bash
Invalid value for option '--providers' (<providers>): expected one of [aws, google, azure] (case-sensitive) but was 'api'

Usage: tw data-links list [OPTIONS]

List data links.

Options:
  -w, --workspace=<workspace>                    Workspace numeric identifier (TOWER_WORKSPACE_ID as default) or workspace reference as
                                                   OrganizationName/WorkspaceName
      --page=<page>                              Pages to display [default: 1].
      --offset=<offset>                          Rows record offset [default: 0].
      --max=<max>                                Maximum number of records to display [default: 100].
  -c, --credentials=<credentialsRef>             Credentials identifier.
      --wait                                     When present program will wait till all data links are fetched to cache.
      --visibility=<visibilityOption>            Show  only data links that are [hidden, visible, all]. When not present all will be shown.
  -n, --name=<startsWith>                        Show only data links with names that start with the given word.
  -r, --region=<region>                          Show only data links belonging to given region
  -p, --providers=<providers>[,<providers>...]   Show only data links belonging to given providers. [aws,azure,google]
  -u, --uri=<uri>                                Show only data links with URI (resource reference) that start with the given URI.
  -h, --help                                     Show this help message and exit.
  -V, --version                                  Print version information and exit.
```